File: StickyNotesJava/app/src/main/java/org/geeksforgeeks/demo/AppWidget.java
package org.geeksforgeeks.demo;

import android.app.PendingIntent;
import android.appwidget.AppWidgetManager;
import android.appwidget.AppWidgetProvider;
import android.content.Context;
import android.content.Intent;
import android.widget.RemoteViews;

// Implementation of App Widget functionality
public class AppWidget extends AppWidgetProvider {

    // Create an Intent to launch activity
    @Override
    public void onUpdate(Context context, AppWidgetManager appWidgetManager, int[] appWidgetIds) {
        // Perform this loop procedure for each App Widget that belongs to this provider
        for (int appWidgetId : appWidgetIds) {
            // Create an Intent to launch MainActivity
            Intent launchActivity = new Intent(context, MainActivity.class);
            launchActivity.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP);

            // Attaching a Pending Intent to trigger it when application is not alive
            PendingIntent pendingIntent = PendingIntent.getActivity(
                    context,
                    0,
                    launchActivity,
                    PendingIntent.FLAG_UPDATE_CURRENT | PendingIntent.FLAG_IMMUTABLE
            );

            // Get the layout for the App Widget and attach an on-click listener to the button
            RemoteViews remoteViews = new RemoteViews(context.getPackageName(), R.layout.app_widget);
            remoteViews.setOnClickPendingIntent(R.id.appwidget_text, pendingIntent);

            // Tell the AppWidgetManager to perform an update on the current app widget
            appWidgetManager.updateAppWidget(appWidgetId, remoteViews);
        }
    }
}

File: StickyNotesKotlin/app/src/main/java/org/geeksforgeeks/demo/AppWidget.kt
package org.geeksforgeeks.demo

import android.app.PendingIntent
import android.appwidget.AppWidgetManager
import android.appwidget.AppWidgetProvider
import android.content.Context
import android.content.Intent
import android.widget.RemoteViews


// Implementation of App Widget functionality
class AppWidget : AppWidgetProvider() {
    // Create an Intent to launch activity
    override fun onUpdate(
        context: Context,
        appWidgetManager: AppWidgetManager,
        appWidgetIds: IntArray
    ) {
        // Perform this loop procedure for each App Widget
        // that belongs to this provider
        for (appWidgetId in appWidgetIds) {
            // Create an Intent to launch MainActivity
            val launchActivity = Intent(
                context,
                MainActivity::class.java
            ).apply {
                addFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TOP)
            }

            // Attaching a Pending Intent
            // to trigger it when
            // application is not alive
            val pendingIntent = PendingIntent.getActivity(context, 0, launchActivity,
                PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE)


            // Get the layout for the App Widget and attach
            // an on-click listener to the button
            val remoteViews = RemoteViews(context.packageName, R.layout.app_widget)
            remoteViews.setOnClickPendingIntent(R.id.appwidget_text, pendingIntent)

            // Tell the AppWidgetManager to perform an
            // update on the current app widget
            appWidgetManager.updateAppWidget(appWidgetId, remoteViews)
        }
    }
}